docs/reference/aggregations/bucket/geohexgrid-aggregation.asciidoc

```diff
@@ -0,0 +1,245 @@
+values into buckets that represent a grid.
+The resulting grid can be sparse and only
+contains cells that have matching data. Each cell corresponds to a
+https://h3geo.org/docs/core-library/h3Indexing#h3-cell-indexp[H3 cell index] a
+and labeled using the https://h3geo.org/docs/core-library/h3Indexing#h3index-representation
+[H3Index representation].
+
+* High precision keys have a larger range for x and y, and represent tiles that
+cover only a small area.
+* Low precision keys have a smaller range for x and y, and represent tiles that
+each cover a large area.
+
+See https://h3geo.org/docs/core-library/restable[the table od cell areas for H3]
+resolutions on how precision (zoom) correlates to size on the ground.
+Precision for this aggregation can be between 0 and 15, inclusive.
+
+WARNING: The highest-precision geohex for precision 15 produces cells that cover
+less than a 10cm by 10cm of land and so high-precision requests can be very
+costly in terms of RAM and result sizes. Please see the example below on how
+to first filter the aggregation to a smaller geographic area before requesting
+high-levels of detail.
+
+You can only use `geohex_grid` to aggregate an explicitly mapped `geo_point`.
+If the `geo_point` field contains an array, `geohex_grid` aggregates all the array values.
+
+
+==== Simple low-precision request
+
+[source,console,id=geohexgrid-aggregation-example]
+--------------------------------------------------
+PUT /museums
+{
+  "mappings": {
+    "properties": {
+      "location": {
+        "type": "geo_point"
+      }
+    }
+  }
+}
+
+POST /museums/_bulk?refresh
+{"index":{"_id":1}}
+{"location": "52.374081,4.912350", "name": "NEMO Science Museum"}
+{"index":{"_id":2}}
+{"location": "52.369219,4.901618", "name": "Museum Het Rembrandthuis"}
+{"index":{"_id":3}}
+{"location": "52.371667,4.914722", "name": "Nederlands Scheepvaartmuseum"}
+{"index":{"_id":4}}
+{"location": "51.222900,4.405200", "name": "Letterenhuis"}
+{"index":{"_id":5}}
+{"location": "48.861111,2.336389", "name": "Musée du Louvre"}
+{"index":{"_id":6}}
+{"location": "48.860000,2.327000", "name": "Musée d'Orsay"}
+
+POST /museums/_search?size=0
+{
+  "aggregations": {
+    "large-grid": {
+      "geohex_grid": {
+        "field": "location",
+        "precision": 4
+      }
+    }
+  }
+}
+--------------------------------------------------
+
+Response:
+
+[source,console-result]
+--------------------------------------------------
+{
+  ...
+  "aggregations": {
+    "large-grid": {
+      "buckets": [
+        {
+          "key": "841969dffffffff",
+          "doc_count": 3
+        },
+        {
+          "key": "841fb47ffffffff",
+          "doc_count": 2
+        },
+        {
+          "key": "841fa4dffffffff",
+          "doc_count": 1
+        }
+      ]
+    }
+  }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/\.\.\./"took": $body.took,"_shards": $body._shards,"hits":$body.hits,"timed_out":false,/]
+
```

Comment:
I'd insert the parameter definitions (currently the `Options` section) here. I'd also add a section immediately after with the response body definitions.

Reply:
Not sure what you are asking here. This is in line with what we have for other grid aggregations so if we change it here we should change for all of them.